### ISSUE TITLE
[FEATURE REQUEST] - Make filters and sorting persistent again

### ISSUE THREAD

**Issue by LnLcFlx** [IgnisDa/ryot#545]
With the frontend changes in `v4`, filters and sorting are no longer persistent.
I really liked persistent filters because I usually only want to see my completed movies in the movies tab and always manually applying the filter is tedious.

I think the best solution would be to specify the default filters and sorting in preferences and then use those if no filters are applied manually.

**Comment by IgnisDa** [comment]
Huh, storing filters in preferences does not sound like a good idea. Going to preferences settings every time you want to set a default is not a good UX, not to mention it would increase the preferences page by a lot.

I thought of a cookie based approach to solving it, but I haven't fleshed out a good solution yet. I'll put some more thought to it once I iron out all the little bugs v4 introduced.

**Comment by lapius** [comment]
You could add a button "Save as default" directly in filter popup. So if you would actually need to change it, no preferences is needed. And for different rarely used filtering you can do it manually.

**Comment by IgnisDa** [comment]
@lapius I am revisiting this issue. Your idea sounds good to me.

So if I understand correctly, the filter will have a btn to save filters. Once that is clicked, these filters will be saved to cookies. Next time the user visits a bare URL (i.e., `/media/movies`), then reload the page with the filters applied. But if the user visits with cookies already applied (i.e., `/media/movies?collectionFilter=whatever`) then the filters wont be respected.

Basically the filters in the url take precedence over the filters in the cookies.

Does this sound like a good approach?

**Comment by IgnisDa** [comment]
Released with `v6.5.0`. Enabled by default and can be disabled in the general preferences.
